code/ui/blocks/src/blocks/useOf.ts

```diff
@@ -31,5 +23,19 @@ export const useOf = (
       )}`
     );
   }
-  return { ...resolved, projectAnnotations: context.projectAnnotations };
+  switch (resolved.type) {
+    case 'component': {
+      return { ...resolved, projectAnnotations: context.projectAnnotations };
+    }
+    case 'meta': {
+      return {
+        ...resolved,
+        preparedMeta: prepareMeta(resolved.csfFile.meta, context.projectAnnotations),
+      };
+    }
+    case 'story':
+    default: {
+      return resolved;
+    }
+  }
```

Comment:
one could argue that this functionality should be directly in `DocsContext.resolveModuleExport` and not in `useOf`.

I'm indifferent.

Reply:
We can always change it later if we need it higher up.